### DISCUSSION TITLE
How can I use a remote nrepl in .nrepl-port?

### DISCUSSION THREAD

**Discussion by agriffis** [Olical/conjure#594] [Q&A] [answered]
Hi! I can connect to my remote nrepl with `:ConjureConnect nrepl://192.168.9.9:7888`

but I can't find any formula for this that works in `.nrepl-port`. All of the following fail:

- `192.168.9.9:7888`
- `nrepl:192.168.9.9:7888`
- `nrepl://192.168.9.9:7888`

Is there a way to tell Conjure about this remote nrepl without needing to explicitly call `:ConjureConnect` every time I start neovim?

**Comment by Olical** [comment]
Conjure only supports a port in the port file, not a full address. I was
under the impression this is how CIDER works in emacs too, if not then I
can probably add support for it.

I don't THINK the port file is supposed to contain anything other than a
port, happy to be proved wrong. Until then maybe a custom Neovim command
local to the repo would work well. I'm fairly sure Conjure supports
overriding the default host too? Could be imagining that but if so, setting
that would allow the port file to take you to the remote host on startup.

**Reply by agriffis** [reply, marked as answer]
Okay, thanks. I created a second file `.nrepl-host` and the following configuration in lazyvim

```lua
  {
    'Olical/conjure',
    lazy = true,
    ft = { 'clojure' },
    config = function(_, opts)
      local _, content = pcall(vim.fn.readfile, '.nrepl-host', '', 1)
      if content then
        vim.g['conjure#client#clojure#nrepl#connection#default_host'] = content[1]
      end
      require('conjure.main').main()
      require('conjure.mapping')['on-filetype']()
    end,
  }
```

I'm also using https://github.com/airblade/vim-rooter so my CWD is always the root of the project, so I don't have to do anything special to find `.nrepl-host`

This will work fine.

**Reply by Olical** [reply]
Ohh this is really cool! Nice work! Could potentially be a cool core addition too. I think it's the right call to keep `.nrepl-port` only containing a port, so this is the next best thing.
